meerk40t/device/ruida/ruidadevice.py

```diff
@@ -1133,6 +1115,11 @@ def process(self, array):
                 # Only seen in Absolute Coords. MachineZero is Ref2 but does not Set Absolute.
         elif array[0] == 0xE7:
             if array[1] == 0x00:
+                if len(self.plotcut):
+                    self.plotcut.settings = self.cutset
+                    self.plotcut.check_if_rasterable()
+                    self.cutcode.append(self.plotcut)
+                    self.plotcut = PlotCut()
```

Comment:
Plotcuts are defined at the blockend. Since this is ends the ruida block and different cut objects should have different blocks and thus different plotcuts.